### PR TITLE
fix(finetune): update Unsloth integration

### PR DESCRIPTION
**Pull request by sozercan**
## Summary

- update the fine-tuning stack to Unsloth and Unsloth Zoo 2026.8.1 with PyTorch 2.10.0 on CUDA 12.6
- migrate the training entrypoint to the current TRL `SFTConfig` and `processing_class` APIs
- embed the matching Python entrypoint in the frontend instead of downloading a potentially mismatched script from `main`
- normalize current Unsloth GGUF output handling so custom output names work
- discover NVIDIA device majors dynamically and include `cmake` for llama.cpp fallback builds
- preserve the public `loadIn4bit` boolean API while applying its documented default and correctly identifying fine-tune configs with omitted optional sections

## Root cause

The previous integration pinned a September 2024 Unsloth commit while allowing modern transitive dependencies to resolve. On August 2, 2026, that produced an incompatible TRL/Transformers stack and failed during import before model loading. The old script also used legacy trainer arguments, downloaded its implementation from an external branch at build time, assumed GGUF files remained under `model/`, and hard-coded NVIDIA UVM device numbers.

## Impact

Fine-tuning builds now use a reproducible current Unsloth stack, execute the script compiled into the frontend, train successfully on current TRL, and export a correctly named GGUF artifact. Explicit `loadIn4bit: false` remains supported while omitted values default to true.

## Validation

All validation was run on `sertac-vm-gpu-amp2` with an NVIDIA A100 80 GB GPU; no local Docker or local builds were used.

- `make test`
- `golangci-lint v2.11.2` — 0 issues
- `python3 -m py_compile pkg/finetune/target_unsloth.py`
- dependency/GPU sentinel build reached the expected missing-model error without import or accelerator failures
- one-step TinyLlama fine-tune with a four-record Alpaca dataset
- GGUF export with a non-default output name
- CUDA inference image build and successful chat-completions request using the generated GGUF
- final independent review reported no remaining concrete findings

### CODE REVIEW

File: pkg/finetune/target_unsloth.py [hunk 1]
```diff
@@ -1,46 +1,50 @@
...
-        data = yaml.safe_load(config_file)
-        print(data)
-    except yaml.YAMLError as exc:
-        print(exc)
 
-cfg = data.get('config').get('unsloth')
-max_seq_length = cfg.get('maxSeqLength')
+with open("/config.yaml", "r", encoding="utf-8") as config_file:
+    data = yaml.safe_load(config_file)
+print(data)
```
[the remark this answers is not in the document]
Comment (sozercan) [reply pkg/finetune/target_unsloth.py]: Fixed in `0df2f95`. The script now emits only the fixed non-sensitive progress message `Loaded fine-tuning configuration.` instead of logging the parsed configuration. I validated the exact pushed commit on `sertac-vm-gpu-amp2` with Python compilation, an AST assertion on the remaining `print` call, and `git diff --check`.

File: pkg/aikit2llb/finetune/convert.go [hunk 2]
```diff
@@ -4,17 +4,26 @@ import (
...
 
 const (
-	unslothCommitOrTag = "fb77505f8429566f5d21d6ea5318c342e8a67991" // September-2024
-	nvidiaMknod        = "mknod --mode 666 /dev/nvidiactl c 195 255 && mknod --mode 666 /dev/nvidia-uvm c 235 0 && mknod --mode 666 /dev/nvidia-uvm-tools c 235 1 && mknod --mode 666 /dev/nvidia0 c 195 0 && nvidia-smi"
-	sourceVenv         = ". .venv/bin/activate"
+	unslothVersion = "2026.8.1"
+	torchVersion   = "2.10.0"
+	sourceVenv     = ". .venv/bin/activate"
+
+	nvidiaMknod = "NVIDIA_MAJOR=$(awk '$2 == \"nvidia\" { print $1; exit }' /proc/devices) && " +
```
[the remark this answers is not in the document]
Comment (sozercan) [reply pkg/aikit2llb/finetune/convert.go]: Fixed in `49f11d8`. NVIDIA primary-device discovery now prefers `nvidia-frontend`, falls back to legacy `nvidia`, and emits an actionable failure if neither is present. Added regression coverage for frontend-only, legacy-only, frontend preference, and missing-device cases. Focused tests, full `make test`, and golangci-lint all passed on `sertac-vm-gpu-amp2`.

File: pkg/aikit/config/finetune_specs.go [hunk 3]
```diff
@@ -9,6 +9,23 @@ type FineTuneConfig struct {
...
+// UnmarshalYAML applies nested fine-tuning defaults even when optional config sections are omitted.
+func (c *FineTuneConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	type plain FineTuneConfig
+
+	decoded := plain{
+		Config: FineTuneConfigSpec{
+			Unsloth: FineTuneConfigUnslothSpec{LoadIn4bit: true},
+		},
+	}
+	if err := unmarshal(&decoded); err != nil {
```
[the remark this answers is not in the document]
Comment (sozercan) [reply pkg/aikit/config/finetune_specs.go]: Fixed in `49f11d8`. The unmarshal path now separately tracks field presence so omitted, null, and empty `config`/`unsloth` sections retain `loadIn4bit: true`, while an explicit `loadIn4bit: false` remains false. Added regression cases for all of those forms. Focused tests and the full `make test` suite passed on `sertac-vm-gpu-amp2`.

**Comment by sozercan** [comment]
@codex review

Please review the latest head (`49f11d8`) after the addressed review feedback.
